data/vibe/data/serialization.d

```diff
@@ -480,10 +479,15 @@ private template serializeValueImpl(Serializer, alias Policy) {
 			ser.serializeValue!(CustomType, ATTRIBUTES)(value.toRepresentation());
 		} else static if (isISOExtStringSerializable!TU) {
 			ser.serializeValue!(string, ATTRIBUTES)(value.toISOExtString());
-		} else static if (isSinkSerializable!TU && isSerializerSupportSinkType!(Serializer, TU)) {
-			ser.serializeSinkType(value);
-		}
-		else static if (isStringSerializable!TU) {
+		} else static if (isStringSinkSerializable!TU) {
+			static if (doesSerializerSupportStringSink!Serializer) {
+				ser.writeStringSinkValue!Traits(value);
+			} else {
+				auto app = appender!string;
+				app.formattedWrite("%s", value);
+				ser.serializeValue!(string, ATTRIBUTES)(app.data);
+			}
+		} else static if (isStringSerializable!TU) {
```

Comment:
I am thinking whether we should change the order of
(line 482) else static if (isStringSinkSerializable!TU) 
(line 490) else static if (isStringSerializable!TU)
for backword compatibility

I think classes might already have toString(sink) type methods, because std functions like formattedwrite already supports them. After this change is merged this sink type toString(sink) will be used instead of the normal toString which might suprises users.

Reply:
Yeah, this might be an issue, although I'd expect in 99% of the cases that both overloads produce identical results and of course they *should* always do that. The fact that turning this around would force the user code to completely remove the string returning overload (if performance is desired) sounds like the worse trade-off to me, because having such an overload for convenience is quite common from my experience.

Reply:
@ferencdg : Having the sink overload take precedence is a well-established practice, actually.
`std.format` does that, and some utilities (e.g. `Throwable.toString`) take advantage of it. Ocean does that too.